lib/aws/lex/conversation/type/slot.rb

```diff
@@ -7,61 +7,82 @@ module Type
         class Slot
           include Base
 
-          required :current_intent, from: :current_intent, virtual: true
-          required :name
-          required :value
-          optional :active, virtual: true
+          required :shape, default: -> { 'Scalar' }
+          optional :name, virtual: true
+          required :lex_value, from: :value, default: -> { {} }, virtual: true
+          required :lex_values, from: :values, default: -> { [] }, virtual: true
+          required :active, default: -> { false }, virtual: true
 
-          def as_json(_opts = {})
-            to_lex
-          end
+          coerce(
+            shape: SlotShape,
+            lex_value: SlotValue,
+            lex_values: Array[Slot]
+          )
 
           def to_lex
-            value
+            super.merge(
+              value: transform_to_lex(lex_value),
+              values: transform_to_lex(lex_values)
+            )
+          end
+
+          def value=(val)
+            return if shape.list?
```

Comment:
I really like the way you handled this API. Use `value` if the slot is defined as a `Scalar`, use `values` if it's defined as a `List`.

The developer should be aware of what type of slot they're dealing with as it needs to be configured on the Lex model.

We could raise an exception (maybe `TypeError`?) rather than returning `nil` so the it's very clear if the developer makes a mistake. What do you think?

Reply:
absolutely. great idea Jesse.